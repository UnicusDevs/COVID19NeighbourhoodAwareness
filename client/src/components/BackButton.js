import React from 'react';
import {Link} from "react-router-dom";

// CSS
import styles from './../sass/components/BackButton.module.scss';

const BackButton = () => {
  return (
    <div className={styles.buttonContainer}>
      <Link to={"/"}><button>Back</button></Link>
    </div>
  )
};

export default BackButton;