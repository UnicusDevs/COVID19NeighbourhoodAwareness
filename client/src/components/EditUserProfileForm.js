import React, {useState, useEffect} from 'react';
import jwt from 'jsonwebtoken'

// Redux
import {connect} from 'react-redux';

// Actions
import { setCurrentUser } from './../redux/actions/userActions';

// API Calls
import { getCurrentUser } from './../api/getUserData';
const UserProfileForm = props => {

  useEffect(() => {
    onLoad();
  }, []);

  async function onLoad() {
    
    let userId = '';

    await getCurrentUser().then((response) => {
      userId = response.data.id;
      jwt.decode(response.config.headers.Authorization);
      console.log(response.data)
      setFirstName(response.data.FirstName);
      setLastName(response.data.LastName);
    })
  };

  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');

  const userName = firstName + ' ' + lastName;

  const handleOnChangeFirstName = (event) => {
    const value = event.target.value;
    setFirstName(value);
  };

  return (
    <form>
      <div>
        <div>
          <h1> {userName} </h1>
        </div>

        <div>
          <div>
            <label> First Name </label>
          </div>

          <input 
            name="firstName"
            value={firstName}
            type="text"
            onChange={handleOnChangeFirstName}
          />
        </div>
      </div>
    </form>
  )
};

// Below calls dispatch with redux store. 
const mapDispatchToProps = (dispatch) => {
  return {
    setCurrentUser: (userData) => dispatch(setCurrentUser(userData)),
  }
};


function mapStateToProps(state) {
  return {
    currentUser: state.userReducer.currentUser
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(UserProfileForm);