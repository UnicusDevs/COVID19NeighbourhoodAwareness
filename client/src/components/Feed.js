import React, {useState} from 'react';

// API Calls
import { getLimitedPosts } from './../api/handlePost';

// Redux 
import { connect } from 'react-redux';
import { addNewPostToAllPostStore } from './../redux/actions/postActions';

// Components
import LogStatus from './LogStatus';
import LogStatusButton from './LogStatusButton';

// Styles
import styles from './../sass/components/Feed.module.scss';

const Feed = (props) => {

  const [pageNumber, setPageNumber] = useState(1);
 
  const onScroll = async (event) => {
    let element = event.target
    if (element.scrollHeight - element.scrollTop === element.clientHeight && props.currentUser === null) {
      setPageNumber(pageNumber + 1)
      await getLimitedPosts(pageNumber).then(async (response) => {
        const newPosts = response.data;
        props.addNewPostToAllPostStore(newPosts)
      }).catch((err) => {
        console.log(err)
      })
    } else if (element.scrollHeight - element.scrollTop === element.clientHeight && props.currentUser) {
      setPageNumber(pageNumber + 1)
      await getLimitedPosts(pageNumber, props.currentUser.id).then(async (response) => {
        const newPosts = response.data;
        props.addNewPostToAllPostStore(newPosts)
      }).catch((err) => {
        console.log(err)
      })
    }
  } 
  
  const handleAllPosts = () => {

    if (props.allPosts === null) {
      return (
        <div>
          <h1> Loading...</h1>
        </div>
      )
    } else if (props.currentUser === null && props.allPosts) {

      let posts = props.allPosts;
      
      posts.sort(function (a, b) {
        // Turn your strings into dates, and then subtract them
        // to get a value that is either negative, positive, or zero.
        return new Date(b.createdAt) - new Date(a.createdAt);
      });

      posts = props.allPosts.map((post) => {

        const userId = post.User;
        const claps = post.Claps;
        const createdAt = post.createdAt;
        const suburb = post.Suburb;
        const postId = post._id;

        return (
          <div key={post._id}>
            <LogStatus postId={postId} user={userId} createdAt={createdAt} suburb={suburb} claps={claps} />
          </div>
        )
      });

      return posts
      // If the use is logged in it will only display the posts in their suburb.
    } else if (props.currentUser && props.filteredPosts.length > 0 ){
      let filteredPosts = props.filteredPosts;

      filteredPosts.sort(function (a, b) {
        // Turn your strings into dates, and then subtract them
        // to get a value that is either negative, positive, or zero.
        return new Date(b.createdAt) - new Date(a.createdAt);
      });

      filteredPosts = props.filteredPosts.map((post) => {
        const userId = post.User;
        const claps = post.Claps;
        const createdAt = post.createdAt;
        const suburb = post.Suburb;
        const postId = post._id;

        return (
          <div key={post._id}>
            <LogStatus postId={postId} user={userId} createdAt={createdAt} suburb={suburb} claps={claps} />
          </div>
        )
      });

      return filteredPosts;
    } else if (props.currentUser && props.filteredPosts.length <= 0) {
      return (
        <div className={styles.noPostsContainer}>
          <h1> Oops... Looks like there are no posts <span role="img">ðŸ˜Ÿ</span>. Be the first to post!</h1>
        </div>
      )
    } else {
      return (
        <div className={styles.noPostsContainer}>
          <h1> Oops... something went wrong <span role="img">ðŸ˜Ÿ</span></h1>
        </div>
      )
    }
  };

  return (
    <div className={styles.feed} onScroll={onScroll} >
      <div className={styles.logFeedContainer}>
        <LogStatusButton />
        {handleAllPosts()}
      </div>
    </div>
  );
};

// Below calls dispatch with redux store. 
const mapDispatchToProps = (dispatch) => {
  return {
    addNewPostToAllPostStore: (newPost) => dispatch(addNewPostToAllPostStore(newPost))
  }
};

function mapStateToProps(state) {
  return {
    currentUser: state.userReducer.currentUser,
    allPosts: state.postReducer.allPosts,
    filteredPosts: state.postReducer.filteredPosts
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(Feed);