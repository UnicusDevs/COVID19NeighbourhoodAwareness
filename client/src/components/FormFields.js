import React from 'react';

export const renderInputField = ({
  input, 
  label, 
  placeholder,
  type,
  meta: { touched, error }
}) => {
  return (
    <div>
      <label> {label} </label>

      <div>
        <input 
          {...input}
          placeholder={placeholder || label}
          type={type}
        />
        {touched && error && <span> {error} </span>}
      </div>
    </div>
  );
};