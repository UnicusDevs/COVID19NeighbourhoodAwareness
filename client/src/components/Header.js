import React, { useEffect } from "react"
import { Link } from 'react-router-dom';

// Redux 
import { connect } from 'react-redux';
import { togglePopUpOnSignUp, togglePopUpOnLogin, togglePopUpOffLogin, togglePopUpOffSignUp } from "./../redux/actions/popUpActions";
import { setCurrentUser } from './../redux/actions/userActions';
import { saveLatestPostDataToStore, saveAllPostsDataToStore } from './../redux/actions/postActions';

//Styles
import styles from '../sass/components/Header.module.scss';

// API Calls
import { getLatestPost, getAllPosts } from './../api/handlePost';
import { getCurrentUser } from './../api/getUserData';
import jwt from 'jsonwebtoken'


const Header = (props) => {

  useEffect(() => {
    onLoad();
  }, []);

  // The below gets the following data: 1. current user 2. currentUsers latest post 3. All posts
  // All data is saved to the redux store.
  async function onLoad() {
    await getCurrentUser().then((response) => {
      jwt.decode(response.config.headers.Authorization)
      props.setCurrentUser(response.data);
      return getLatestPost(response.data.id)
    }).then((response) => {
      const latestPostData = response.data;
      props.saveLatestPostDataToStore(latestPostData)
    }).catch((err) => {
      props.saveLatestPostDataToStore(null)
      props.setCurrentUser(null);
    })

    getAllPosts().then(async (response) => {
      const allPostsData = response.data;
      await props.saveAllPostsDataToStore(allPostsData)
    }).catch((err) => {
      props.saveAllPostsDataToStore(null)
    })
  };

  const handleSignOut = () => {
    const now = new Date()
    now.setTime(now.getTime() - 1)
    document.cookie = `covid19Project=;expires=${now.toUTCString()};path=/`;
    props.setCurrentUser(null);
    return window.location.reload();
  }

  // The below handles the pop up for signup. More can be found in the signup Form.
  const handleTogglePopUpOnSignUp = () => {
    props.togglePopUpOffLogin()
    props.togglePopUpOnSignUp()
  };

  // The below handles the pop up for the login. More can be found in the login form.
  const handleTogglePopUpOnLogin = () => {
    props.togglePopUpOffSignUp()
    props.togglePopUpOnLogin()
  };

  const handleHeaderDisplay = (currentUser) => {
    if (!props.currentUser) {
      return (
        <header>
          <div className={styles.container}>
            <div className={styles.innercontainer}>
              <div className={styles.logo}>
                <Link to="/">Home2Home</Link>
              </div>
              <div className={styles.navigation}>
                <nav>
                  <button onClick={handleTogglePopUpOnLogin}>Log in</button>
                  <button onClick={handleTogglePopUpOnSignUp}>Sign up</button>
                </nav>
              </div>
            </div>
          </div>
        </header>
      )
    } else {
      return (
        <header>
          <div className={styles.container}>
            <div className={styles.innercontainer}>
              <div className={styles.logo}>
                <Link to="/">Home2Home</Link>
              </div>
              <div className={styles.navigation}>
                <nav>
                  <button onClick={handleSignOut}>Sign out</button>
                </nav>
              </div>
            </div>
          </div>
        </header>
      )
    }
  };

  return (
    <div>
      {handleHeaderDisplay(props.currentUser)}
    </div>
  )
};

// Below calls dispatch with redux store. 
const mapDispatchToProps = (dispatch) => {
  return {
    togglePopUpOnSignUp: () => dispatch(togglePopUpOnSignUp()),
    togglePopUpOffSignUp: () => dispatch(togglePopUpOffSignUp()),
    togglePopUpOnLogin: () => dispatch(togglePopUpOnLogin()),
    togglePopUpOffLogin: () => dispatch(togglePopUpOffLogin()),
    setCurrentUser: (userData) => dispatch(setCurrentUser(userData)),
    saveLatestPostDataToStore: (latestPostData) => dispatch(saveLatestPostDataToStore(latestPostData)),
    saveAllPostsDataToStore: (allPostsData) => dispatch(saveAllPostsDataToStore(allPostsData))
  }
};

function mapStateToProps(state) {
  return {
    currentUser: state.userReducer.currentUser
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);