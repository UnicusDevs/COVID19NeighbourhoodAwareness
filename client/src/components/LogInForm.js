import React from 'react';
import Cookies from 'universal-cookie';

// React hook
import { useForm } from 'react-hook-form';

// Axios / cookies
import axiosAPI from "./../api/baseURL";

// CSS
import styles from './../sass/components/SignupForm.module.scss';

// Redux 
import {connect} from 'react-redux';
import { togglePopUpOffLogin } from "./../redux/actions/popUpActions";

const cookies = new Cookies();

let LoginForm = props => {

  // To Do: Add token on login stage
  // The below is a axios post to create new user then log them in. 
  let sendUserToDatabase = (values) => {
    axiosAPI.post("/login", {
        EmailAddress: values.emailAddress,
        Password: values.password
      }).then(response => {
        // Below sets the token. To view more of the token go to baseURL.js
        const token = response.data;
        cookies.set("covid19Project", token, { path: "/" })
        window.location.assign("/");
      }).catch((err) => {
    })
  };

  const {register, handleSubmit, errors} = useForm({
    validateCriteriaMode: "all",
    mode: "onSubmit"
  });

  // The below sends the data off to the store, and calls axios function
  const onSubmit = formData => {
    // Below calls axios function
    sendUserToDatabase(formData);
  };

  const handlePopUpClose = () => {
    props.togglePopUpOffLogin()
  }

  return (
    <form onSubmit={handleSubmit(onSubmit)} className={styles.form} >
      <div className={styles.signUpFormContainer}>

        <div className={styles.inputContainer}>
          <div>
            <label> Email Address </label>
          </div>

          <input 
            name="emailAddress" 
            placeholder="me@example.com" 
            type="email" 
            ref={register({ required: true, pattern: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/  })} 
          />

          {errors.emailAddress && errors.emailAddress.types.required && (<p>Email required</p>)}
          {errors.emailAddress && errors.emailAddress.types.pattern && (<p>Not a valid email address</p>)}
        </div>

        <div className={styles.inputContainer}>
          <div>
            <label> Password </label>
          </div>

          <input
            name="password"
            placeholder="******"
            type="password"
            ref={register({ required: true })}
          />

          {errors.password && errors.password.types.required && (<p>Password required</p>)}
        </div>

        <div className={styles.buttons}>
          <button onClick={handlePopUpClose}> Cancel </button>
          <button type="submit"> Submit </button>
        </div>

      </div>      
   </form>
  );
};

// Below calls dispatch with redux store. 
const mapDispatchToProps = (dispatch) => {
  return {
    togglePopUpOffLogin: () => dispatch(togglePopUpOffLogin()),
  }
};

function mapStateToProps(state) {
  return {
    displayPopUpLogin: state.popUpReducer.displayPopUpLogin
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);