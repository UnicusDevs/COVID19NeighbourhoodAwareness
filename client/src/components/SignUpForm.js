import React, {useState} from 'react';

// Google API
import GooglePlacesAutocomplete from 'react-google-places-autocomplete';

// React hook
import { useForm } from 'react-hook-form';

// API
import { signup } from "./../api/registration";
import axiosAPI from './../api/baseURL';

// Redux
import { connect } from 'react-redux';
import { saveFormData, saveFormErrorMessages } from "./../redux/actions/signupFormActions.js";

// CSS
import styles from './../sass/components/SignupForm.module.scss';

//Other
import ProfileImageDefault from './../assets/icons8-login-as-user-96.png';

export let SignUpForm = props => {

  const [profileImage, setProfileImage] = useState(ProfileImageDefault);
  const [address, setAddress] = useState("");
  const [file, setFile] = useState({});
  const [fileUrl, setFileUrl] = useState("");

  // The below is a axios post to create new user then log them in. 
  let sendUserToDatabase = async (values) => {

    // The below uploads file to s3 Bucket.
    let fileUrl; 
    const formData = new FormData();
    formData.append('file', file)

    await axiosAPI.post('/signup/upload', formData).then((response) => {
      fileUrl = response.data
    });

    signup(values, fileUrl)
  };

  const {register, handleSubmit, errors} = useForm({
    validateCriteriaMode: "all",
    mode: "onSubmit"
  });

  // The below sends the data off to the store, and calls axios function
  const onSubmit = async (formData, file) => {

    // Below saves formData to redux
    props.saveFormData(formData)
    // Below calls axios function
    sendUserToDatabase(formData) ;
  };

  const handleFileChosen = (event) => {
    const file = event.target.files[0]
    setFile(event.target.files[0])
    // The below creates an object so we can view the image
    setProfileImage(URL.createObjectURL(file));
  };

  const handleChange = (event) => {
    setAddress(event.target.value)
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)} className={styles.form} >
      <div className={styles.signUpFormContainer}>
        <div className={styles.signUpFormHeader}>
          <h1>Sign up</h1>
        </div> 
        <div className={styles.profileImageContainer}>

          <div>
            <label> Profile Image </label>
          </div>
          
          <img src={profileImage} alt="profile" className={styles.profileImage}/>

          <input
            name="profileImage"
            label="Profile Image"
            type="file"
            onChange={handleFileChosen}
            ref={register}
          />

        </div>
        <div className={styles.row}>

          <div className={styles.left}>
            <div className={styles.inputContainer}>
              <div>
                <label> First Name </label>
              </div>

              <input 
                name="firstName" 
                placeholder="Josephine" 
                type="text" 
                ref={register({required: true, minLength: 2})}
              />

              {errors.firstName && errors.firstName.types.required && (<h5>First Name required</h5>)}
              {errors.firstName && errors.firstName.types.minLength && (<h5>Name must be greater than two letters</h5>)}
            </div>

            <div className={styles.inputContainer}>
              <div>
                <label> Last Name </label>
              </div>

              <input
                name="lastName"
                placeholder="Smith"
                type="text"
                ref={register({ required: true, minLength: 2 })}
              />

              {errors.lastName && errors.lastName.types.required && (<h5>Last Name required</h5>)}
              {errors.lastName && errors.lastName.types.minLength && (<h5>last must be greater than two letters</h5>)}
            </div>

            <div className={styles.inputContainer}>
              <div>
                <label> Age </label>
              </div>

              <input 
                name="age" 
                placeholder="24" 
                type="number" 
                ref={register({ required: true })} 
              />

              {errors.age && errors.age.types.required && (<h5>Age required</h5>)}
            </div> 

          <div className={styles.inputContainer}>
            <div>
              <label> Suburb </label>
            </div>
              <GooglePlacesAutocomplete
    
                autocompletionRequest={{
                  componentRestrictions: {
                    country: ['au'],
                  },
                }}

                renderInput={(props) => (
                  <div className={styles.googleInputContainer}>
                    <input
                      name="suburb"
                      type="text"
                      {...props}
                      className={styles.googleInput}
                      ref={register({ required: true, minLength: 2 })}
                    />
                  </div>
                )}

                renderSuggestions={(active, suggestions, onSelectSuggestion) => (
                  <div className={styles.suggestionsContainer}>
                    {suggestions.map((suggestion) => {
                      return (
                        <div
                          key={suggestion.id}
                          className={styles.suggestion}
                          onClick={(event) => onSelectSuggestion(suggestion, event)}
                        >
                          <h4>{suggestion.description}</h4>
                          <hr />
                        </div>
                      )
                    })
                    }
                  </div>
                )}
                />
              {errors.suburb && errors.suburb.types.required && (<h5>Suburb required</h5>)}
            </div>
          </div> 
          <div className={styles.right}>
            <div className={styles.inputContainer}>
              <div>
                <label> Email Address </label>
              </div>

              <input 
                name="emailAddress" 
                placeholder="me@example.com" 
                type="email" 
                ref={register({ required: true, pattern: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/  })} 
              />

              {errors.emailAddress && errors.emailAddress.types.required && (<h5>Email required</h5>)}
              {errors.emailAddress && errors.emailAddress.types.pattern && (<h5>Not a valid email address</h5>)}
            </div>

            <div className={styles.inputContainer}>
              <div>
                <label> Password </label>
              </div>

              <input
                name="password"
                placeholder="******"
                type="password"
                ref={register({ required: true })}
              />

              {errors.password && errors.password.types.required && (<h5>Password required</h5>)}
            </div>

            <div className={styles.inputContainer}>   
              <div>
                <label> Confirm Password </label>
              </div>

              <input
                name="confirmPassword"
                placeholder="******"
                type="password"
                ref={register({ required: true })}
              /> 

              {errors.confirmPassword && errors.confirmPassword.types.required && (<h5>Please confirm your password</h5>)}
            </div>
            <div className={styles.buttons}>
              <button type="submit"> Submit </button>
            </div>
          </div>
        </div>
      </div>      
   </form>
  );
};

// Below calls dispatch with redux store. 
const mapDispatchToProps = (dispatch) => {
  return {
    saveFormData: (formData) => dispatch(saveFormData(formData)),
    saveFormErrorMessages: (errorMessage) => dispatch(saveFormErrorMessages(errorMessage)),
  }
};

function mapStateToProps(state) {
  return {
    displayPopUpSignUp: state.popUpReducer.displayPopUpSignUp
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(SignUpForm);