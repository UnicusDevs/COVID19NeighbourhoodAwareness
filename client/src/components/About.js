import React from 'react';
import Feed from './../components/Feed';
import styles from '../sass/components/About.module.scss';

const About = (props) => {
	return (
		<div className={styles.about}>
			<div className={styles.container}>
				<div className={styles.row}>
					<div className={styles.left}>
						<p className={styles.title}>Stay apart. Remain united</p>
						<p>The fight against COVID-19 depends on each of us doing the right thing. But in isolation it can be hard to see or feel the good we’re doing. PROJECT NAME is designed to change that. Through the simple daily click of a button, users can let their neighbours know they’re acting to protect the community. By visiting the page, they can easily visualise, and be inspired by, the amazing and caring efforts of those around them. The only way to beat this virus is by working together. Log your fight against COVID-19 today!
						</p>
					</div>
					<div className={styles.right}>
						<Feed />
					</div>
				</div>
			</div>
		</div>
	)
}

export default About;