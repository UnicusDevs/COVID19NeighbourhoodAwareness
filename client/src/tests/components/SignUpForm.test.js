import React from 'react';
import { shallow } from 'enzyme';
import {SignUpForm} from './../../components/SignUpForm';
import userData from './../mockData/userData';
import toJson from 'enzyme-to-json';


test('Should render SignUpForm correctly', () => { 
  const component = shallow(<SignUpForm />);
  expect(toJson(component)).toMatchSnapshot();
});

// test('Should throw validation errors', () => {

//   const component = shallow(
//     <SignUpForm />
//   );

//   component.find('form').simulate('submit', {
//     preventDefault: () => {}
//   });

//   expect(component.find({errors}).length).toBeGreaterThan(0);
//   expect(component).toMatchSnapshot();
// });

test('Should change on first name', () => {
  const component = shallow(<SignUpForm />);
  const value = 'Joe';
  
  component.find('input').at(1).simulate('change', {target: {value: value}})

  expect(component).toBe(value);
})