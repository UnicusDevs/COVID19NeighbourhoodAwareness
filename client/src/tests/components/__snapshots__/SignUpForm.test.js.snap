// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SignUpForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="signUpFormContainer"
  >
    <div
      className="signUpFormHeader"
    >
      <h1>
        Sign up
      </h1>
    </div>
    <div
      className="profileImageContainer"
    >
      <img
        alt="profile"
        className="profileImage"
        src=""
      />
      <div>
        <label>
           Profile Image 
        </label>
      </div>
      <input
        label="Profile Image"
        name="profileImage"
        onChange={[Function]}
        type="file"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="left"
      >
        <div
          className="inputContainer"
        >
          <div>
            <label>
               First Name 
            </label>
          </div>
          <input
            name="firstName"
            placeholder="Josephine"
            type="text"
          />
        </div>
        <div
          className="inputContainer"
        >
          <div>
            <label>
               Last Name 
            </label>
          </div>
          <input
            name="lastName"
            placeholder="Smith"
            type="text"
          />
        </div>
        <div
          className="inputContainer"
        >
          <div>
            <label>
               Age 
            </label>
          </div>
          <input
            name="age"
            placeholder="24"
            type="number"
          />
        </div>
        <div
          className="inputContainer"
        >
          <div>
            <label>
               Suburb 
            </label>
          </div>
          <GooglePlacesAutocomplete
            apiKey=""
            autocompletionRequest={
              Object {
                "componentRestrictions": Object {
                  "country": Array [
                    "au",
                  ],
                },
              }
            }
            debounce={300}
            disabled={false}
            idPrefix=""
            initialValue=""
            inputClassName=""
            inputStyle={Object {}}
            loader={null}
            onSelect={[Function]}
            placeholder="Address..."
            renderInput={[Function]}
            renderSuggestions={[Function]}
            required={false}
            suggestionsClassNames={
              Object {
                "container": "",
                "suggestion": "",
                "suggestionActive": "",
              }
            }
            suggestionsStyles={
              Object {
                "container": Object {},
                "suggestion": Object {},
              }
            }
            withSessionToken={false}
          />
        </div>
      </div>
      <div
        className="right"
      >
        <div
          className="inputContainer"
        >
          <div>
            <label>
               Email Address 
            </label>
          </div>
          <input
            name="emailAddress"
            placeholder="me@example.com"
            type="email"
          />
        </div>
        <div
          className="inputContainer"
        >
          <div>
            <label>
               Password 
            </label>
          </div>
          <input
            name="password"
            placeholder="******"
            type="password"
          />
        </div>
        <div
          className="inputContainer"
        >
          <div>
            <label>
               Confirm Password 
            </label>
          </div>
          <input
            name="confirmPassword"
            placeholder="******"
            type="password"
          />
        </div>
        <div
          className="buttons"
        >
          <button
            type="submit"
          >
             Submit 
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Should render SignupForm correctly with user data 1`] = `ShallowWrapper {}`;
